rtcdtlstransport.go

```diff
@@ -1,13 +1,166 @@
 package webrtc
 
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pions/dtls/pkg/dtls"
+	"github.com/pions/webrtc/internal/mux"
+	"github.com/pions/webrtc/pkg/rtcerr"
+)
+
 // RTCDtlsTransport allows an application access to information about the DTLS
 // transport over which RTP and RTCP packets are sent and received by
 // RTCRtpSender and RTCRtpReceiver, as well other data such as SCTP packets sent
 // and received by data channels.
 type RTCDtlsTransport struct {
-	// Transport RTCIceTransport
+	lock sync.RWMutex
+
+	iceTransport *RTCIceTransport
+	certificates []RTCCertificate
 	// State     RTCDtlsTransportState
 
 	// OnStateChange func()
 	// OnError       func()
+
+	conn *dtls.Conn
+}
+
+// NewRTCDtlsTransport creates a new RTCDtlsTransport.
+// This constructor is part of the ORTC API. It is not
+// meant to be used together with the basic WebRTC API.
+func NewRTCDtlsTransport(transport *RTCIceTransport, certificates []RTCCertificate) (*RTCDtlsTransport, error) {
+	t := &RTCDtlsTransport{iceTransport: transport}
+
+	if len(certificates) > 0 {
+		now := time.Now()
+		for _, x509Cert := range certificates {
+			if !x509Cert.Expires().IsZero() && now.After(x509Cert.Expires()) {
```

Comment:
It worries me doing security stuff like this, I don't feel confident we catch all the edge cases. So if we start catching some, I think people will assume we caught everything.

Before we did nothing, so easy enough to say `please check things yourself` but we can talk about this further outside the PR!